### PR TITLE
set template variable by clicking self link

### LINKED ISSUE #9259: Table panel: Update current dashboard template variable via cell links
- Grafana v4.6.0

To reproduce:

1. Set a table-column to `Render value as link`

1. Create a link that points to the current dashboard but with changed variables (e.g. `https://example.com/dashboard/db/overview?var-server=$__cell`)

1. Try to click the link. The address bar changes but the page does not reload.

### PR DESCRIPTION
**Pull request by mtanda**
Fix https://github.com/grafana/grafana/issues/9259

### CODE REVIEW

File: public/app/core/directives/value_select_dropdown.ts [hunk 1]
```diff
@@ -296,6 +301,40 @@ export function valueSelectDropdown($compile, $window, $timeout, $rootScope) {
         scope
       );
 
+      scope.$on('$locationChangeStart', (event, next) => {
```
Comment (torkelo) [inline public/app/core/directives/value_select_dropdown.ts]: This feels very wrong to me, that this input component should listen for changes like this.

**Review by torkelo — CHANGES_REQUESTED** [review]
This approach feels very bad. Think it will require a bigger change, maybe moving some variable state / variableSrv stuff to redux and update state on location change events.

File: public/app/core/directives/value_select_dropdown.ts (hunk 1 above)
Comment (mtanda) [reply public/app/core/directives/value_select_dropdown.ts]: Sorry, I'm not familiar with redux, if there are good example in Grafana, would you tell me?

Comment (mtanda) [reply public/app/core/directives/value_select_dropdown.ts]: https://github.com/grafana/grafana/blob/master/public/app/core/services/bridge_srv.ts#L41-L56
Seems related code.

Comment (mtanda) [reply public/app/core/directives/value_select_dropdown.ts]: And this?
https://github.com/grafana/grafana/blob/master/style_guides/redux.md

**Comment by torkelo** [comment]
An alternative approach would be for variableSrv to listen to url query changes. 
Example for a way to listen to this via redux location state is: 
https://github.com/grafana/grafana/blob/master/public/app/features/playlist/playlist_srv.ts#L80

**Pull request closed without merging**